### PR TITLE
PCLOUDS-1623-LivenessProbe_failing

### PR DESCRIPTION
**Pull request by joaquinfilipic-dynatrace**
New thread with new loop for webserver, for both Metrics and Logs modes

### CODE REVIEW

File: src/run_docker.py [hunk 1]
```diff
@@ -157,32 +156,24 @@ async def sleep_until_next_polling(current_polling_duration_s):
...
+
     print_dynatrace_logo()
 
     logging_context.log("GCP Monitor - Dynatrace integration for Google Cloud Platform monitoring\n")
 
-    webserver.setup_webserver_on_asyncio_loop(loop, HEALTH_CHECK_PORT)
-
-    instance_metadata = loop.run_until_complete(run_instance_metadata_check())
-    loop.run_until_complete(import_self_monitoring_dashboards(instance_metadata))
+    instance_metadata = asyncio.run(run_instance_metadata_check())
```
Comment (mmajcher) [inline src/run_docker.py:169]: Please make sure that if this check fails then exception is propagated and fails main execution. (assuming this was previous behaviour, at least from my memory)

File: src/run_docker.py [hunk 2]
```diff
@@ -157,32 +156,24 @@ async def sleep_until_next_polling(current_polling_duration_s):
...
-    finally:
-        print("Closing AsyncIO loop...")
-        webserver.close_and_cleanup(loop)
-        loop.close()
-
-
 def main():
+    threading.Thread(target=run_webserver_on_asyncio_loop_forever,
+                     args=(loop,),
+                     name="AioHttpLoopWaiterThread",
```
Comment (mmajcher) [inline src/run_docker.py:162]: WebserverThread

File: src/run_docker.py [hunk 3]
```diff
@@ -157,32 +156,24 @@ async def sleep_until_next_polling(current_polling_duration_s):
...
-        loop.run_forever()
-    finally:
-        print("Closing AsyncIO loop...")
-        webserver.close_and_cleanup(loop)
-        loop.close()
-
-
 def main():
+    threading.Thread(target=run_webserver_on_asyncio_loop_forever,
+                     args=(loop,),
```
Comment (mmajcher) [inline src/run_docker.py:161]: Why share your loop? your = from this thread

File: src/lib/webserver/webserver.py [hunk 4]
```diff
@@ -2,25 +2,34 @@
...
 
-def setup_webserver_on_asyncio_loop(loop, port):
-    global APP, RUNNER
-    logging_context.log("Setting up webserver... \n")
 
-    APP = web.Application()
-    APP.add_routes([web.get('/health', health_endpoint)])
+def run_webserver_on_asyncio_loop_forever(loop):
+    try:
+        global APP, RUNNER
```
Comment (mmajcher) [inline src/lib/webserver/webserver.py:17]: No need for global vars anymore

File: src/run_docker.py (hunk 3 above)
Comment (joaquinfilipic-dynatrace) [reply src/run_docker.py]: Created new loop in thread

File: src/run_docker.py (hunk 1 above)
Comment (joaquinfilipic-dynatrace) [reply src/run_docker.py:168]: Yes, it maintains that

**Review by mmajcher — APPROVED** [review]

**Pull request merged**